autorest/codegen/__init__.py

```diff
@@ -135,12 +139,37 @@ def _build_package_dependency() -> Dict[str, str]:
             "dependency_msrest": "msrest>=0.6.21",
         }
 
+    @staticmethod
+    def _build_security(yaml_data: Dict[str, Any]) -> Dict[str, Any]:
+        security_yaml = yaml_data.get("security")
+        security: Dict[str, Any] = dict()
+        if security_yaml and security_yaml.get("authenticationRequired"):
+            # the key word is "AADToken" and "AzureKey"
+            aad_token = set()
+            azure_key = ""
+            for scheme in security_yaml.get("schemes"):
+                if _AAD_TOKEN == scheme["type"]:
+                    aad_token.update(scheme["scopes"])
+                elif _AZURE_KEY == scheme["type"]:
+                    # only accept the last one
+                    azure_key = scheme["headerName"]
+            if aad_token:
+                security[_AAD_TOKEN] = list(aad_token)
+            if azure_key:
+                security[_AZURE_KEY] = azure_key
+        return security
+
     def _create_code_model(self, yaml_data: Dict[str, Any], options: Dict[str, Union[str, bool]]) -> CodeModel:
         # Create a code model
 
         code_model = CodeModel(options=options)
+        security = self._build_security(yaml_data)
         if code_model.options['credential']:
             self._handle_default_authentication_policy(code_model)
+        elif security:
+            code_model.options['credential'] = True
+            self._handle_authentication_policy(code_model, security)
```

Comment:
do you need to add this new function, can you use [the existing code](https://github.com/Azure/autorest.python/blob/961cddb8dad10d9425c36ffcc4a4f7ab2c8c45da/autorest/codegen/models/credential_schema_policy.py#L67) / refactor it? don't want code duplication

Reply:
`self._handle_default_authentication_policy` is for credential flags and `self._handle_authentication_policy` is for security definition. We shall avoid using them together, so it is better to split the logic.